index.js

```diff
@@ -954,11 +955,23 @@ class XiaomiRoborockVacuum {
         // Delete then add, to maintain the correct order.
         this.roomIdsToClean.delete(roomId);
         this.roomIdsToClean.add(roomId);
+        if (this.config.roomTimeout > 0) {
+          this.log.info(`ACT setCleaningRoom | ${this.model} | Start timeout to clean rooms`);
+          clearTimeout(this._roomTimeout)
+          this._roomTimeout = setTimeout(this.setCleaning.bind(this, true), this.config.roomTimeout * 1000)
+        }
       } else if (!state && !this.isCleaning && !this.isPaused) {
         this.log.info(
           `ACT setCleaningRoom | ${this.model} | Disable cleaning Room ID ${roomId}.`
         );
         this.roomIdsToClean.delete(roomId);
+        if (this.config.roomTimeout > 0) {
+          this.log.info(`ACT setCleaningRoom | ${this.model} | Start timeout to clean rooms`);
+          clearTimeout(this._roomTimeout)
+          if (this.roomIdsToClean.size > 0) {
+            this._roomTimeout = setTimeout(this.setCleaning.bind(this, true), this.config.roomTimeout * 1000)
+          }
+        }
```

Comment:
Do you mind moving this piece of logic to a function and to call it on both branches? If feels like repeating the logic :)